backend/flowpilot/builder.go

```diff
@@ -137,13 +138,39 @@ func (fb *defaultFlowBuilder) validate() error {
 	if len(fb.errorStateName) == 0 {
 		return errors.New("fixed state 'errorState' is not set")
 	}
-	if !fb.flow.stateExists(fb.initialStateName) {
-		return errors.New("fixed state 'initialState' does not belong to the flow")
+	if !fb.flow.stateExists(fb.initialStateName) && !fb.subFlows.stateExists(fb.initialStateName) {
+		return errors.New("fixed state 'initialState' does not belong to the flow or a sub-flow")
 	}
 	if !fb.flow.stateExists(fb.errorStateName) {
 		return errors.New("fixed state 'errorState' does not belong to the flow")
 	}
 
+	// Validate the specified next states, when the flow starts with a sub-flow.
+	if err := fb.validateNextStateNames(); err != nil {
+		return fmt.Errorf("failed to validate the specified next states: %w", err)
+	}
+
+	return nil
```

Comment:
Would be good to validate that the initialState is part of a subFlow.

Reply:
Validation has been improved, see [245daad](https://github.com/teamhanko/hanko/pull/1188/commits/245daad7fb33d64a83bb092a7eea0b05713d3f77)